src/mococrw/asymmetric_encryption.h

```diff
@@ -0,0 +1,90 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#pragma once
+
+#include "key.h"
+#include "openssl_wrap.h"
+#include <type_traits>
+#include "iostream"
+#include "padding_mode.h"
+
+namespace mococrw {
+
+/**
+ * This class ... TODO: complete
+ */
+class AsymmetricEncryption
+{
+public:
+    AsymmetricEncryption() = delete;
+
+private:
+    std::vector<uint8_t> message;
+
+public:
+    class CryptoData
+    {
+    public:
+        CryptoData() = default;
+        CryptoData(const std::string& data);
```

Comment:
Perhaps this class shouldn't accept strings. If the encryption output contains a null terminator character, it could be "cut off" when saved in this format.

Reply:
The user may not be aware of this limitation. Assuming that the intention of using a CryptoData object instead of plain vectors is to simplify the interface for the user, then it shouldn't allow the user to receive incorrect encryption data when using the string format. I'm in favor of just returning std::vector<uint8_t>.

Reply:
Ok, encrypt and decrypted methods modified. 
Encrypt receives a CryptoData and returns a vector<uint8_t> 
Decrypt receives a vector<uint8_t> and returns a CryptoData.